commons/src/main/java/org/calrissian/accumulorecipes/commons/transform/KeyToTupleCollectionQueryXform.java

```diff
@@ -71,35 +69,28 @@ public V apply(Map.Entry<Key, Value> keyValueEntry) {
         eventFields.read(kryo, new Input(keyValueEntry.getValue().get()), EventFields.class);
         V entry = buildTupleCollectionFromKey(keyValueEntry.getKey());
         for (Map.Entry<String, EventFields.FieldValue> fieldValue : eventFields.entries()) {
-            if (selectFields == null || selectFields.contains(fieldValue.getKey())) {
-                String[] aliasVal = splitPreserveAllTokens(new String(fieldValue.getValue().getValue()), ONE_BYTE);
-                Object javaVal = typeRegistry.decode(aliasVal[0], aliasVal[1]);
-
-                String vis = fieldValue.getValue().getVisibility().getExpression().length > 0 ? new String(fieldValue.getValue().getVisibility().getExpression()) : "";
-
-                try {
-                    Collection<Map<String,Object>> meta = metadataSerDe.deserialize(fieldValue.getValue().getMetadata());
-                    if(meta != null) {
-                        for(Map<String,Object> curMeta : meta) {
-                            Map<String,Object> metadata = new HashMap<String,Object>();
-                            if(curMeta != null)
-                                metadata.putAll(curMeta);
-
-                            setVisibility(metadata, vis);
-                            Tuple tuple = new Tuple(fieldValue.getKey(), javaVal, metadata);
-                            entry.put(tuple);
-                        }
-                    } else {
-                        Map<String,Object> metadata = new HashMap<String,Object>();
-                        setVisibility(metadata, vis);
-                        Tuple tuple = new Tuple(fieldValue.getKey(), javaVal, metadata);
-                        entry.put(tuple);
-                    }
-                } catch(Exception e) {
-                    log.error("There was an error deserializing the metadata for a tuple", e);
+            String[] aliasVal = splitPreserveAllTokens(new String(fieldValue.getValue().getValue()), ONE_BYTE);
+            Object javaVal = typeRegistry.decode(aliasVal[0], aliasVal[1]);
+
+            String vis = fieldValue.getValue().getVisibility().getExpression().length > 0 ? new String(fieldValue.getValue().getVisibility().getExpression()) : "";
+
+            try {
+                Map<String,Object> meta = metadataSerDe.deserialize(fieldValue.getValue().getMetadata());
+                if(meta != null) {
```

Comment:
Instead of duplicating most of the logic, why not simply use the if else logic for creating the metadata object a the rest of the logic in this if statement is exactly the same.  For example

Map<String,Object> metadata = (meta == null : new HashMap<String, Object>() ? new HashMap<String, Object>(meta));
setVisibility(metadata, vis);
Tuple tuple = new Tuple(fieldValue.getKey(), javaVal, metadata);
entry.put(tuple);